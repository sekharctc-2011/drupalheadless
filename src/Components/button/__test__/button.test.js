import React from "react"
import ReactDOM from "react-dom"
import ButtonComponent from "./../button"

import { render, cleanup } from "@testing-library/react"
import "@testing-library/jest-dom"

import renderer from "react-test-renderer"

afterEach(cleanup)
it("render without crash", () => {
  const div = document.createElement("div")
  ReactDOM.render(<ButtonComponent></ButtonComponent>, div)
})

it("render button correctly", () => {
  const { getByTestId } = render(<ButtonComponent label="Click here"></ButtonComponent>)
  expect(getByTestId("button")).toHaveTextContent("Click here")
})

it("render button correctly", () => {
  const { getByTestId } = render(<ButtonComponent label="Click me please"></ButtonComponent>)
  expect(getByTestId("button")).toHaveTextContent("Click me please")
})

it("matches snapshort", () => {
  const tree = renderer.create(<ButtonComponent label="Save"></ButtonComponent>).toJSON()
  expect(tree).toMatchSnapshot()
})
