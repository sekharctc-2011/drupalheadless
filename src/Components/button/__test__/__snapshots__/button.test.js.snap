// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshort 1`] = `
<div
  className="btn btn-sm btn-primary"
  data-testid="button"
>
  Hi 
  Save
</div>
`;
